### ISSUE TITLE
Getting different accuracy while using SFS and using classifier with these features.

### ISSUE THREAD

**Issue by mas15** [rasbt/mlxtend#356]
Hello, I have got a dataset with about 700 features. Using Weka I was able to do feature selection and select 120 features that gave me good accuracy 55%.
Then I used SFS on the same dataset:

```
 model = MultinomialNB()
 sfs1 = SFS(model,
               k_features=(50,150),
               forward=True,
               n_jobs=-1,
               floating=False,
               verbose=2,
               scoring='accuracy',
               cv=5)

  sfs1 = sfs1.fit(x, y)
 print(sfs1.k_feature_idx_)
```
And it selected also about 120 features and gave 55% accuracy on the best features set.
But when I use these selected features and train a MultinomialNB I get much worse accuracy - 45%.
Is the accuracy the sfs gave me made on a training set?

**Comment by rasbt** [comment]
Hi, there,

yes the performance of the SFS is based on the average cross-validation performance on the training set. How did you evaluate the MultinomialNB performance? Is this the test set performance?
